Content.Server/GameObjects/Components/NodeNetwork/INodeGroupFactory.cs

```diff
@@ -0,0 +1,75 @@
+﻿using Robust.Shared.Interfaces.Reflection;
+using Robust.Shared.IoC;
+using System;
+using System.Collections.Generic;
+
+namespace Content.Server.GameObjects.Components.NodeGroup
+{
+    public interface INodeGroupFactory
+    {
+        void Initialize();
+
+        INodeGroup MakeNodeGroup(NodeGroupID nodeGroupType);
+    }
+
+    public class NodeGroupFactory : INodeGroupFactory
+    {
+        private readonly Dictionary<NodeGroupID, Type> _groupTypes = new Dictionary<NodeGroupID, Type>();
+
+#pragma warning disable 649
+        [Dependency] private readonly IReflectionManager _reflectionManager;
+        [Dependency] private readonly IDynamicTypeFactory _typeFactory;
+#pragma warning restore 649
+
+        public void Initialize()
+        {
+            var nodeGroupTypes = _reflectionManager.GetAllChildren<NodeGroup>();
+            foreach (var nodeGroupType in nodeGroupTypes)
+            {
+                var att = (NodeGroupAttribute) Attribute.GetCustomAttribute(nodeGroupType, typeof(NodeGroupAttribute));
```

Comment:
You need `System.Reflection`, but this can be changed to `nodeGroupType.GetCustomAttribute<NodeGroupAttribute>()`, I believe.